### PR TITLE
Move KVP-ADD/MODIFY/REMOVE-KEY-VALUES under KVP-VALUES-OPERATIONS

### LINKED ISSUE #689: kvp tests should be independent
Currently, the following test cases: KVP-ADD-KEY-VALUES, KVP-MODIFY-KEY-VALUES and KVP-REMOVE-KEY-VALUES need to be run sequentially on the same VM. This breaks the pattern of independent tests.

They can be all merged as stages in the same test case.

### PR DESCRIPTION
**Pull request by asofron**
Fixes #689 
Consolidated the below 3 testcases into a single one, as all operations are carried one after another : 
KVP-ADD-KEY-VALUES
KVP-MODIFY-KEY-VALUES
KVP- REMOVE-KEY-VALUES

Verification:
```
[LISAv2 Test Results Summary]
Test Run On           : 02/04/2019 14:00:53
VHD Under Test        : ubuntu_18.04.1.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:3

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 KVP-KEY-VALUES-OPERATIONS                                                         PASS                 1.15 
	ADD KVP VALUE : PASS 
	MODIFY KVP VALUE : PASS 
	REMOVE KVP VALUE : PASS 


[LISAv2 Test Results Summary]
Test Run On           : 02/04/2019 14:13:10
VHD Under Test        : Centos_7.5_x64.vhdx
Total Test Cases      : 1 (1 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:3

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 KVP-KEY-VALUES-OPERATIONS                                                         PASS                 1.13 
	ADD KVP VALUE : PASS 
	MODIFY KVP VALUE : PASS 
	REMOVE KVP VALUE : PASS 


```

### CODE REVIEW

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+    $result = $vmManagementService.ModifyKvpItems($vmGuest, $msvmKvpExchangeDataItem.PSBase.GetText(1))
+    $job = [wmi]$result.Job
+
+    # Check if the modify worked
+    while ($job.jobstate -lt 7) {
+        $job.get()
+    }
+
+    if ($job.ErrorCode -ne 0) {
+        Write-LogErr "Error: while modifying the key value pair"
```
[the remark this answers is not in the document]
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Fixed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+        $VMPort,
+        $VMUserName,
+        $VMPassword,
+        $TestParams
+    )
+
+    $key = $null
+    $ModValue = $null
+
+    if (-not $TestParams) {
```
[the remark this answers is not in the document]
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Fixed

File: Testscripts/Linux/KVP-AddKeyValues.sh [hunk 3]
```diff
@@ -4,8 +4,6 @@
 
 # Description:
```
[the remark this answers is not in the document]
Comment (asofron) [reply Testscripts/Linux/KVP-AddKeyValues.sh]: Created SwitchKVPKey Function

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+        Write-LogErr "Unable to add KVP value to guest"
+        Write-LogErr "       error code $($job.ErrorCode)"
+        return $False
+    }
+    if ($job.Status -ne "OK") {
+        Write-LogErr "KVP add job did not complete with status OK"
+        return $False
+    }
+
+    $KVPAddKeyValues = SwitchKVPKey -VMUserName $VMUserName -VMPassword $VMPassword -Ipv4 $Ipv4 -VMPort $VMPort -Key $Key -Pool $Pool -Value $AddValue
```
Comment (chvalean) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: line too long

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+    if (-not $KVPAddKeyValues) {
+        Write-LogErr "Failed to modifiy KVP item"
```
Comment (chvalean) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: modify

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+    }
+
+    if ($job.Status -ne "OK") {
+        Write-LogErr "KVP modify job did not complete with status OK"
+        return $False
+    }
+
+    $KVPModifyKeyValues = SwitchKVPKey -VMUserName $VMUserName -VMPassword $VMPassword -Ipv4 $Ipv4 -VMPort $VMPort -Key $Key -Pool $Pool -Value $ModValue
+    if (-not $KVPModifyKeyValues) {
+        Write-LogErr "Failed to modifiy KVP item"
```
Comment (chvalean) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: modify

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+}
+function Main {
+    param([object] $allVMData, [object] $CurrentTestData, [object] $TestProvider, [string]$TestParams)
+
+    if (-not $TestParams) {
+        Write-LogErr "No TestParams provided"
+        Write-LogErr "       This script requires the Key & value test parameters"
+        return "Aborted"
+    }
+    # Find the TestParams we require.  Complain if not found
```
Comment (chvalean) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: replace with
Find the TestParams required

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+param([object] $AllVmData, [object] $CurrentTestData, [object] $TestProvider, [string]$TestParams)
+
+
+# Define KVP operations in functions
+function AddKVPKey {
+    param (
+        $VMName,
+        $HvServer,
+        $RootDir,
+        $Ipv4,
```
Comment (chvalean) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: not required

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 9]
```diff
@@ -0,0 +1,405 @@
...
+        Write-LogInfo "KVP item successfully modified"
+        return $True
+    }
+}
+function ModifyKVPKey {
+    param (
+        $VMName,
+        $HvServer,
+        $RootDir,
+        $Ipv4,
```
Comment (chvalean) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: not required

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 10]
```diff
@@ -0,0 +1,405 @@
...
+
+param([object] $AllVmData, [object] $CurrentTestData, [object] $TestProvider, [string]$TestParams)
+
+
+# Define KVP operations in functions
+function AddKVPKey {
+    param (
+        $VMName,
+        $HvServer,
+        $RootDir,
```
Comment (ader1990) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: remove unused params from the function definition

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 11]
```diff
@@ -0,0 +1,405 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+# This script handles the following KVP operations:
+# - Add a KVP key and value
```
Comment (ader1990) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: be more specific on which side host/guest? is the step made

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 4 above)
Comment (ader1990) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: line too long

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 5 above)
Comment (ader1990) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: you mean failed to add?

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 12]
```diff
@@ -0,0 +1,405 @@
...
+        return $False
+    }
+
+    $KVPAddKeyValues = SwitchKVPKey -VMUserName $VMUserName -VMPassword $VMPassword -Ipv4 $Ipv4 -VMPort $VMPort -Key $Key -Pool $Pool -Value $AddValue
+    if (-not $KVPAddKeyValues) {
+        Write-LogErr "Failed to modifiy KVP item"
+        return $False
+    }
+    else {
+        Write-LogInfo "KVP item successfully modified"
```
Comment (ader1990) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: ditto

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 [hunk 13]
```diff
@@ -0,0 +1,405 @@
...
+    if ($job.Status -ne "OK") {
+        Write-LogErr "KVP delete job did not complete with status OK"
+        return $False
+    }
+
+    # If we made it here, everything worked
+    Write-LogInfo "KVP item successfully deleted"
+    return $True
+}
+Function SwitchKVPKey {
```
Comment (ader1990) [inline Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: use Get-Verb for this method definition. 

the name of this method should be what is this method doing.

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 8 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Remov

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 4 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Fixed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 6 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Fixed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 7 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Fixed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 8 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Removed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 9 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Removed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 10 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: removed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 5 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Fixed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 12 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Fixed

File: Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1 (hunk 4 above)
Comment (asofron) [reply Testscripts/Windows/KVP-KEY-VALUES-OPERATIONS.ps1]: Fixed

**Comment by asofron** [comment]
> remove from lisav2 hyperv the usage of HyperVApplyCheckpoint flag.

Removed

**Pull request merged**